src/main/java/com/yahoo/identity/services/key/KeyService.java

```diff
@@ -1,5 +1,11 @@
 package com.yahoo.identity.services.key;
 
+import javax.annotation.Nonnull;
+
 public interface KeyService {
 
+    @Nonnull
+    String getSecret();
+
+    void setSecret(@Nonnull String secret);
```

Comment:
Why we need this method? 
I suggest you to just read the secret from a file.
And `getSecret` should take a `String name`, so that caller can indicate which secret it wants to get.

Reply:
I change it to read from file now. If a username is needed to retrieve the key, what about the secret to verify token in TokenApiService where the username is not passed as a parameter?